zetaclient/evm/base_transaction_data_test.go

```diff
@@ -0,0 +1,85 @@
+package evm
+
+import (
+	"math/big"
+	"testing"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+	corecommon "github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+func TestSigner_SetChainAndSender(t *testing.T) {
+	// setup inputs
+	cctx, err := getCCTX()
+	require.NoError(t, err)
+
+	txData := &BaseTransactionData{}
+	logger := zerolog.Logger{}
+
+	t.Run("SetChainAndSender PendingRevert", func(t *testing.T) {
+		cctx.CctxStatus.Status = types.CctxStatus_PendingRevert
+		skipTx := txData.SetChainAndSender(cctx, logger)
+
+		require.False(t, skipTx)
+		require.Equal(t, ethcommon.HexToAddress(cctx.InboundTxParams.Sender), txData.to)
+		require.Equal(t, big.NewInt(cctx.InboundTxParams.SenderChainId), txData.toChainID)
+	})
+
+	t.Run("SetChainAndSender PendingOutBound", func(t *testing.T) {
+		cctx.CctxStatus.Status = types.CctxStatus_PendingOutbound
+		skipTx := txData.SetChainAndSender(cctx, logger)
+
+		require.False(t, skipTx)
+		require.Equal(t, ethcommon.HexToAddress(cctx.GetCurrentOutTxParam().Receiver), txData.to)
+		require.Equal(t, big.NewInt(cctx.GetCurrentOutTxParam().ReceiverChainId), txData.toChainID)
+	})
+
+	t.Run("SetChainAndSender Should skip cctx", func(t *testing.T) {
+		cctx.CctxStatus.Status = types.CctxStatus_PendingInbound
+		skipTx := txData.SetChainAndSender(cctx, logger)
+		require.True(t, skipTx)
+	})
+}
+
+func TestSigner_SetupGas(t *testing.T) {
+	cctx, err := getCCTX()
+	require.NoError(t, err)
+
+	evmSigner, err := getNewEvmSigner()
+	require.NoError(t, err)
+
+	txData := &BaseTransactionData{}
+	logger := zerolog.Logger{}
+
+	t.Run("SetupGas_success", func(t *testing.T) {
+		chain := corecommon.BscMainnetChain()
+		err := txData.SetupGas(cctx, logger, evmSigner.EvmClient(), &chain)
+		require.NoError(t, err)
+	})
+
+	t.Run("SetupGas_error", func(t *testing.T) {
+		cctx.GetCurrentOutTxParam().OutboundTxGasPrice = "invalidGasPrice"
+		chain := corecommon.BtcMainnetChain()
+		err := txData.SetupGas(cctx, logger, evmSigner.EvmClient(), &chain)
+		require.Error(t, err)
```

Comment:
I have been curious whether the dynamic gas price special handling for Ethereum chain is still valid or not. We can confirm with Tanmay or Bertrand in future if we need to retain that logic. But you could use `require.ErrorContains(t, err, "cannot convert gas price")` to more accurately hit that error path. Also, using a EVM chain, e.g. `BscMainnetChain()`, will probably help us to better understand this test case.

Reply:
I think that may have been a typo, should be BscMainnetChain()